### ISSUE TITLE
How to specify that one field is used as the length for two distinct slices?

### ISSUE THREAD

**Issue by mewmew** [go-restruct/restruct#25]
The file format I've been trying to parse uses a single `Length` field to specify the length of two distinct slices. However, I've been unable to model this using restruct.

Using the following code did not seem to work:

```go
struct {
	Frames uint32 `struct:"sizeof=FrameSizes,sizeof=FrameTypes"`
	FrameSizes []uint32
	FrameTypes []uint32
}
```

`FrameTypes` is parsed correctly, but `FrameSizes` results in an empty slice.

```
    FrameSizes:   nil,
    FrameTypes:   {0x58, 0x82, 0x0, 0x0, 0xc, 0xa, 0x0, 0x0, 0xf0, 0x9, 0x0, 0x0, 0x38, 0xa, 0x0, 0x0, 0x58, 0xa, 0x0,  ...},
```

Any suggestions? Is it or should it be possible to use reflect for this purpose?

**Comment by jchv** [comment]
Never really considered this. I kind of like the approach of reversing things, though. I think the solution to this could be similar to the solution to #23 in some way. Perhaps we could specify a size key, that can tie one integer field to specify the size of one or more slices or strings. This isn't perfect; it wouldn't handle a case where you have an array of sizes specifying the lengths for arrays in an array of arrays. But I think it would come closer to handling more challenging cases. Interested in feedback on how we could make handling of variable size arrays as flexible and intuitive as possible.

**Comment by mewmew** [comment]
As a reference, a `sizefrom` tag was added to `struc` in https://github.com/lunixbochs/struc/issues/55. This would fix #25, but not #23. Not quite sure how to do that in a good way (implementation wise), but code wise it could use a solution similar to `sizefrom`:

```go
type Header struct {
	Frames uint32
}

struct {
	Hdr Header
	FrameSizes []uint32 `struct:"sizefrom=Hdr.Frames"`
	FrameTypes []uint32 `struct:"sizefrom=Hdr.Frames"`
}
```

Something along those lines.

**Comment by jchv** [comment]
I know it's a bit late, but I'm finally implementing sizefrom support, though not with the ability to reference other structures. The main reason I haven't been implementing the latter is because Restruct automatically pulls the length of a slice back into the sizeof fields.
